app/(auth)/login/page.tsx

```diff
@@ -1,51 +1,29 @@
 'use client';
 
 import { useState } from 'react';
-import { useRouter } from 'next/navigation';
-import supabase from '../../../api/supabase/createClient';
+import { AuthProvider, useAuth } from '../../utils/AuthProvider';
 
-export default function Login() {
+export default function LoginLayout() {
+  return (
+    <AuthProvider>
+      <Login />
+    </AuthProvider>
+  );
+}
+
+function Login() {
+  const { signIn } = useAuth();
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
-  const { push } = useRouter();
 
-  const handleSignUp = async () => {
-    const { data, error } = await supabase.auth.signUp({
-      email,
-      password,
-    });
-
-    if (error) {
-      // Check if the error is due to an already registered email
-      if (error.message.includes('User already registered')) {
-        throw new Error(
-          'This email is already registered. Please try signing in instead.',
-        );
-      } else {
-        throw new Error(
-          `An error occurred trying to sign up: ${error.message}`,
-        );
+  const handleSignIn = async () => {
+    try {
+      await signIn(email, password);
+    } catch (error) {
+      if (error instanceof Error) {
+        console.error(error.message);
       }
     }
-
-    push('/');
-
-    return data;
-  };
-
-  const handleSignInWithEmail = async () => {
-    const { data, error } = await supabase.auth.signInWithPassword({
-      email,
-      password,
-    });
-
-    if (error) {
-      throw new Error(`An error occurred trying to sign in: ${error}`);
-    }
-
-    push('/');
-
-    return data;
   };
 
   return (
```

Comment:
we may want to make the outermost container a `form` to improve the accessibility of this page. 

here's some skeleton code from chatGPT 
```
import React, { useState } from "react";

export default function Signup() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [errorMessage, setErrorMessage] = useState("");

  const isFormValid = email && password && confirmPassword && password === confirmPassword;

  const handleSignUp = (e) => {
    e.preventDefault(); // Prevent default form submission behavior

    // Clear error message and proceed with sign-up logic
    setErrorMessage("");
    // Add your sign-up logic here
  };

  return (
    <form onSubmit={handleSignUp}>
      <input
        name="email"
        onChange={(e) => setEmail(e.target.value)}
        value={email}
        placeholder="Email"
        required
      />
      <input
        type="password"
        name="password"
        onChange={(e) => setPassword(e.target.value)}
        value={password}
        placeholder="Password"
        required
      />
      <input
        type="password"
        name="confirmPassword"
        onChange={(e) => setConfirmPassword(e.target.value)}
        value={confirmPassword}
        placeholder="Confirm Password"
        required
      />
      {errorMessage && <p style={{ color: "red" }}>{errorMessage}</p>}
      <button type="submit" disabled={!isFormValid}>
        Sign up
      </button>
    </form>
  );
}

export default SignUpForm
```